astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -457,8 +457,8 @@ def __init__(
         self.name = name
         """The name of the module."""
 
-        self.doc = doc
-        """The module docstring."""
+        # TODO: Remove this attribute
+        self._doc = doc
```

Comment:
At least for `Module.__init__` `doc` is still a required argument. I think we can add a default value.